pkg/config/expand.go

```diff
@@ -176,14 +178,16 @@ func useModule(
 			continue
 		}
 
-		// track whether setting already has a value; important when
-		// ensuring that the first used module takes precedence and also
-		// when flattening multiple values for settings that are lists
+		// if the input exists in the module doing the using, then we modify it
+		// according to these rules:
+		// if the setting has a value, then it was set by "use" because we have
+		// skipped anything explicitly in the blueprint. If the setting is a
+		// list of any type, then we want to concatenate these values in order
+		// they were used; otherwise, the setting should remain set to the
+		// existing used value
 		_, isAlreadySet := mod.Settings[settingName]
```

Comment:
Strictly speaking this would be true for how the function ought to be used, but it might not be the case if one supplies an empty `settingsInBlueprint`. I'll consider a change for clarification.

Reply:
I have changed it to `setByUse`